src/main/java/net/rptools/maptool/model/framework/LibraryToken.java

```diff
@@ -162,9 +163,16 @@ private static Library findLibrary(URL path) {
       List<Token> tokensFiltered =
           zone.getTokensFiltered(t -> name.equalsIgnoreCase(t.getName())).stream()
               .filter(Token::getAllowURIAccess)
-              .collect(Collectors.toList());
+              .toList();
       if (tokensFiltered.size() > 0) {
         return new LibraryToken(tokensFiltered.get(0).getId());
+      } else { // Check to see if URI access is not allowed and throw error
+        List<Token> tokens =
+            zone.getTokensFiltered(t -> name.equalsIgnoreCase(t.getName())).stream().toList();
```

Comment:
Since we only check whether or not the list is empty, there's no need to convert to a stream and back.

Also, we could save the unmodifiable list above (the one returned by the first call to `getTokensFiltered()`) and avoid regenerating it here with repeated logic.

Reply:
You are correct I was trying to get it done in a hurry over lunch and fell afoul of the dreaded cut and paste/modify slightly curse. I will update after work tonight.